pilot/pkg/model/endpointshards.go

```diff
@@ -404,22 +404,22 @@ func updateShardServiceAccount(shards *EndpointShards, serviceName string) bool
 	return false
 }
 
-// EndpointIndexUpdater is an updater that will keep an EndpointIndex in sync. This is intended for tests only.
-type EndpointIndexUpdater struct {
+// TestEndpointIndexUpdater is an updater that will keep an EndpointIndex in sync. This is intended for tests only.
+type TestEndpointIndexUpdater struct {
 	Index *EndpointIndex
 	// Optional; if set, we will trigger ConfigUpdates in response to EDS updates as appropriate
 	ConfigUpdateFunc func(req *PushRequest)
 }
 
-var _ XDSUpdater = &EndpointIndexUpdater{}
+var _ XDSUpdater = &TestEndpointIndexUpdater{}
 
-func NewEndpointIndexUpdater(ei *EndpointIndex) *EndpointIndexUpdater {
-	return &EndpointIndexUpdater{Index: ei}
+func NewEndpointIndexUpdater(ei *EndpointIndex) *TestEndpointIndexUpdater {
```

Comment:
maybe a step further add a separate file to hold all tests helpers. 
Kubernetes has many fake.go fake_xx.go, play the same goal

Reply:
SG. Changed. PTAL